include/aws/common/encoding.h

```diff
@@ -0,0 +1,96 @@
+#ifndef AWS_COMMON_ENCODING_H
+#define AWS_COMMON_ENCODING_H
+
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/common.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * Base 16 (hex) encodes the contents of to_encode and stores the result in output.
+ * If output is NULL, output_size will be set to what the output_size should be.
+ */
+AWS_COMMON_API int aws_hex_encode(const uint8_t *to_encode, size_t to_encode_len, char *output, size_t *output_size);
+
+/*
+ * Base 16 (hex) decodes the contents of to_decode and stores the result in output.
+ * If output is NULL, output_size will be set to what the output_size should be.
+ */
+AWS_COMMON_API int aws_hex_decode(const char *to_decode, size_t to_decode_len, uint8_t *output, size_t *output_size);
+
+/*
+ * Base 64 encodes the contents of to_encode and stores the result in output.
```

Comment:
Should we support urlsafe b64?

Reply:
what in particular would you be wanting in that case? base64 that has been url encoded? In that case no.